packages/nextjs/src/performance/client.ts

```diff
@@ -1,13 +1,104 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-import { Primitive, Transaction, TransactionContext } from '@sentry/types';
-import { fill, getGlobalObject, stripUrlQueryAndFragment } from '@sentry/utils';
+import { getCurrentHub } from '@sentry/hub';
+import { Primitive, TraceparentData, Transaction, TransactionContext } from '@sentry/types';
+import {
+  extractTraceparentData,
+  fill,
+  getGlobalObject,
+  logger,
+  parseBaggageHeader,
+  stripUrlQueryAndFragment,
+} from '@sentry/utils';
+import type { NEXT_DATA as NextData } from 'next/dist/next-server/lib/utils';
 import { default as Router } from 'next/router';
+import type { ParsedUrlQuery } from 'querystring';
 
 const global = getGlobalObject<Window>();
 
 type StartTransactionCb = (context: TransactionContext) => Transaction | undefined;
 
+/**
+ * Describes data located in the __NEXT_DATA__ script tag. This tag is present on every page of a Next.js app.
+ */
+interface SentryEnhancedNextData extends NextData {
+  // contains props returned by `getInitialProps` - except for `pageProps`, these are the props that got returned by `getServerSideProps` or `getStaticProps`
+  props: {
+    _sentryGetInitialPropsTraceData?: string; // trace parent info, if injected by server-side `getInitialProps`
+    _sentryGetInitialPropsBaggage?: string; // baggage, if injected by server-side `getInitialProps`
+    pageProps?: {
+      _sentryGetServerSidePropsTraceData?: string; // trace parent info, if injected by server-side `getServerSideProps`
+      _sentryGetServerSidePropsBaggage?: string; // baggage, if injected by server-side `getServerSideProps`
+    };
+  };
+}
+
+interface NextDataTagInfo {
+  route?: string;
+  traceParentData?: TraceparentData;
+  baggage?: string;
+  params?: ParsedUrlQuery;
+}
+
+/**
+ * Every Next.js page (static and dynamic ones) comes with a script tag with the id "__NEXT_DATA__". This script tag
+ * contains a JSON object with data that was either generated at build time for static pages (`getStaticProps`), or at
+ * runtime with data fetchers like `getServerSideProps.`.
```

Comment:
Given that `getStaticProps` runs an arbitrary amount of time before the page is served (both when it's run at build time and when it's run at runtime as part of revalidation), it doesn't make sense to send trace data from it, only the parameterized route. Do we want to note that here somehow? Could be in the docstring here or could be below where you're dealing with the other two, to explain why it's not all three.

Reply:
I finally read up on how ISR works and I think I fully understand it now. There is one case where it totally makes sense to trace `getStaticProps` and I went ahead and added logic and an explanation for it: 652d3eb5a4c7341fcf8939430e52086793200a34